internal/apis/bootstrap/kubeadm/v1alpha3/conversion.go

```diff
@@ -77,6 +77,10 @@ func (src *KubeadmConfig) ConvertTo(dstRaw conversion.Hub) error {
 		}
 		dst.Spec.JoinConfiguration.Patches = restored.Spec.JoinConfiguration.Patches
 		dst.Spec.JoinConfiguration.SkipPhases = restored.Spec.JoinConfiguration.SkipPhases
+
+		if restored.Spec.JoinConfiguration.Discovery.File != nil && restored.Spec.JoinConfiguration.Discovery.File.KubeConfig != nil {
+			dst.Spec.JoinConfiguration.Discovery.File.KubeConfig = restored.Spec.JoinConfiguration.Discovery.File.KubeConfig
```

Comment:
Is it possible that dst.Spec.JoinConfiguration.Discovery.File is nil? (I know the fuzz-tests don't catch it, but that doesn't mean it can't happen in reality)

(same below of course)

(same for v1alpha4)

Reply:
Added it, in reality though (I think) this can't happen and I didn't add it because for `restored.Spec.JoinConfiguration.Discovery.File` to be not nil, the `KubeConfigPath` string will be set in both places

Reply:
I think it can be nil at least if a client sets it to nil during a round trip (not saying it's likely, just possible 😃)